File: backend/events.go
package main

type ErrorEvent struct {
	Event string `json:"event"`
	Error string `json:"error"`
}

type CreateEvent struct {
	Event string `json:"event"`
	Alias string `json:"alias"`
}

type JoinEvent struct {
	Event string `json:"event"`
	Id    string `json:"id"`
}

type DisconnectEvent struct {
	Id         string `json:"id"`
	Event      string `json:"event"`
	ChatroomId string `json:"chatroom_id"`
}

type DisconnectedEvent struct {
	Event string `json:"event"`
}

type MessageSendEvent struct {
	Id         string `json:"id"`
	Event      string `json:"event"`
	Message    string `json:"message"`
	ChatroomId string `json:"chatroom_id"`
}

type MessageReceivedEvent struct {
	Event   string `json:"event"`
	Message string `json:"message"`
}

File: mobile/app/src/main/java/com/dutchaen/kandichat/models/Events.kt
package com.dutchaen.kandichat.models

import com.google.gson.Gson

interface Event {
    val event: String
}

data class ErrorEvent(
    override val event: String = "error",
    val error: String
) : Event

data class CreateEvent(
    override val event: String = "create",
    val alias: String = "Anonymous"
) : Event

data class JoinEvent(
    override val event: String = "join",
    val id: String
) : Event

data class MessageSendEvent(
    val id: String,
    override val event: String = "message_send",
    val message: String,
    val chatroom_id: String,
) : Event

data class MessageReceivedEvent(
    override val event: String = "message_receive",
    val message: String,
) : Event

data class DisconnectEvent(
    val id: String,
    override val event: String = "disconnect",
    val chatroom_id: String,
) : Event

fun Event.asJsonPayload(): String {
    val gson = Gson()
    return gson.toJson(this);
}